### PR TITLE
more verbose logging for model import/export

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- python/amici/logging.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4801957&bid=15980201)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- python/amici/logging.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4801957&bid=15980201)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- python/amici/logging.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4801957&bid=15980201)

File: python/amici/logging.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import socket
+import amici
+import os
+import warnings
+import time
+import functools
+
+LOG_LEVEL_ENV_VAR = 'AMICI_LOG'
+BASE_LOGGER_NAME = 'amici'
+NAMED_LOG_LEVELS = {'NOTSET': logging.NOTSET,
```
Comment (dweindl) [inline python/amici/logging.py]:
```suggestion
# Supported values for LOG_LEVEL_ENV_VAR
NAMED_LOG_LEVELS = {'NOTSET': logging.NOTSET,
```

File: python/amici/logging.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+        console_output: Set up a default console log handler if True (default)
+
+        file_output: Supply a filename to copy all log output to that file, or
+            set to False to disable (default)
+
+        capture_warnings: Capture warnings from Python's warnings module
+            if True (default)
+    Returns:
+        A logging.Logger object for PySB logging. Note that other PySB modules
```
Comment (dweindl) [inline python/amici/logging.py]:
```suggestion
        A logging.Logger object for AMICI logging. Note that other AMICI modules
```

File: python/amici/logging.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    log.debug('Python version: %s', platform.python_version())
+    log.debug('Hostname: %s', socket.getfqdn())
+
+    logging.captureWarnings(capture_warnings)
+
+    return log
+
+
+def get_logger(logger_name=BASE_LOGGER_NAME, log_level=None, **kwargs):
+    """ Returns (if extant) or creates an AMICI logger
```
Comment (dweindl) [inline python/amici/logging.py]:
```suggestion
    """ Returns (if extistant) or creates an AMICI logger
```

File: python/amici/logging.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    If the AMICI base logger has already been set up, this method will
+    return it or any of its descendant loggers without overriding the
+    settings - i.e. any values supplied as kwargs will be ignored.
+    Parameters:
```
Comment (dweindl) [inline python/amici/logging.py]:
```suggestion

    Parameters:
```

**Comment by FFroehlich** [comment]
> Great!
> 
> Might be very helpful to set the log level in one of the examples notebooks.

Added to steadystate example

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- python/amici/logging.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4801957&bid=15980201)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- python/amici/logging.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4801957&bid=15980201)

**Pull request merged**